how-to/customize-workspace/client/src/framework/integrations.ts

```diff
@@ -198,8 +208,18 @@ export async function getHelpSearchEntries(): Promise<HomeSearchResult[]> {
 
 	for (const integrationModule of integrationModules) {
 		if (integrationModule.isInitialised && integrationModule.implementation.getHelpSearchEntries) {
-			const integrationResults = await integrationModule.implementation.getHelpSearchEntries();
-			results = results.concat(integrationResults);
+			const helpSearchEntries = await integrationModule.implementation.getHelpSearchEntries();
+
+			if (integrationHelpers?.setSearchQuery) {
+				for (const helpEntry of helpSearchEntries) {
```

Comment:
Is this done so that help entries populate the search string in a consistent way without having to learn the api? I am guessing an integration has access to the helper function and could implement their own version if they wanted to?

Reply:
Yes exactly that, help entries just need to set the populateQuery data property then everything else is handled for them. But through the helpers they can call setSearchQuery directly if they want to in their own actions